### DISCUSSION TITLE
Zigbee2qmtt e ZHA possono coesistere?

### DISCUSSION THREAD

**Discussion by asmaran85** [indomus/forum#2080] [HUB: Home Assistant]
Buongiorno,
ho parecchi dispositivi zigbee perfettamente configurati in ZHA e poi ho 7 dispositivi zigbee tuya che non vengono correttamente riconosciuti da ZHA e che utilizzo mediante l'integrazione tuya.
Posso utilizzare zigbee2qmtt per i 7 dispositivi tuya senza disinstallare zha?
I due sistemi possono coesistere sfruttando entrambi lo stesso coordinator?
Grazie

**Comment by snakuzzo** [comment]
No...i coordinator devono essere diversi

**Comment by asmaran85** [comment]
Grazie, esiste un modo per migrare tutti i dispositivi da ZHA a zigbee2qmtt mantenendo lo stesso nome?

**Reply by snakuzzo** [reply]
Sinceramente non ho mai provato e non mi pare sia documentato un passaggio "indolore" (concedimi il termine) da ZHA a zigbee2mqtt.
Credo soprattutto per un discorso legato ai parametri della network che si andrà a creare. Andrebbe minuziosamente creata allo stesso modo, ma resterebbe comunque la gestione da parte del software, che però è completamente diversa.

Quindi... NO ... non puoi farlo.

Per quanto riguarda il nome...
- o cancelli prima tutte le entità legate a ZHA e poi quando fai il pairing con Z2M gli dai lo stesso nome "sperando" che alla fine esca fuori lo stesso
- oppure fai il pairing e poi man mano li adatti

IMHO è tempo sprecato. Per quanto possano essere numerosi i device, con un editor/IDE decente (vedi VS Code o similari) fai un find/replace in tutti i file YAML da te creati e nelle dashboard e hai finito

**Reply by snakuzzo** [reply]
P.S. Quando qui fai reply, fallo sotto l'ultimo messaggio e non creare alti thread come hai fatto poco fa.
Grazie

**Reply by asmaran85** [reply]
Ok grazie
